src/Microsoft.TemplateEngine.Edge/Installers/NuGet/NugetApiPackageManager.cs

```diff
@@ -98,75 +104,114 @@ public async Task<DownloadResult> DownloadPackageAsync(InstallRequest installReq
                 SourceRepository repository = Repository.Factory.GetCoreV3(source);
                 try
                 {
-                    PackageMetadataResource resource = await repository.GetResourceAsync<PackageMetadataResource>().ConfigureAwait(false);
+                    PackageMetadataResource resource = await repository.GetResourceAsync<PackageMetadataResource>(cancellationToken).ConfigureAwait(false);
                     resources[source] = resource;
                 }
                 catch (Exception e)
                 {
-                    _nugetLogger.LogError($"Failed to load the NuGet source {source}");
-                    _nugetLogger.LogDebug($"Reason: {e.Message}");
+                    _nugetLogger.LogError($"Failed to load the NuGet source {source}.");
+                    _nugetLogger.LogDebug($"Reason: {e.Message}.");
                 }
             }
 
             if (!resources.Any())
             {
-                throw new InvalidNuGetSourceException("Failed to load NuGet sources", attemptedSources);
+                throw new InvalidNuGetSourceException("Failed to load NuGet sources", packageSources.Select(source => source.Source));
             }
 
-            ConcurrentDictionary<string, IEnumerable<IPackageSearchMetadata>> searchResults = new ConcurrentDictionary<string, IEnumerable<IPackageSearchMetadata>>();
-            foreach (KeyValuePair<string, PackageMetadataResource> resource in resources)
+            Dictionary<PackageSource, IEnumerable<IPackageSearchMetadata>> searchResults = new Dictionary<PackageSource, IEnumerable<IPackageSearchMetadata>>();
+            foreach (KeyValuePair<PackageSource, PackageMetadataResource> resource in resources)
             {
-                IEnumerable<IPackageSearchMetadata> foundPackages = await resource.Value.GetMetadataAsync(
-                    packageIdentifier,
-                    includePrerelease: false,
-                    includeUnlisted: false,
-                    cache,
-                    _nugetLogger,
-                    cancellationToken).ConfigureAwait(false);
-                if (foundPackages.Any())
+                try
                 {
-                    _nugetLogger.LogVerbose($"Found {foundPackages.Count()} {packageIdentifier} packages in NuGet feed {resource.Key}");
-                    searchResults[resource.Key] = foundPackages;
+                    IEnumerable<IPackageSearchMetadata> foundPackages = await resource.Value.GetMetadataAsync(
+                        packageIdentifier,
+                        includePrerelease: false,
+                        includeUnlisted: false,
+                        cache,
+                        _nugetLogger,
+                        cancellationToken).ConfigureAwait(false);
+
+                    if (foundPackages.Any())
+                    {
+                        _nugetLogger.LogDebug($"Found {foundPackages.Count()} {packageIdentifier} packages in NuGet feed {resource.Key}.");
```

Comment:
Finding multiple packages of the same id might be a bit confusing to the reader. All these are different versions, right? How about
```
Found 3 different versions of the package "{packageIdentifier}" in NuGet feed "{resource.Key}".
```

Or, maybe just getting rid of the count would be better.

Reply:
Rephrased as `Found {foundPackages.Count()} versions for {packageIdentifier} in NuGet feed {resource.Key}."